cmd/osm-bootstrap/crds/config_meshconfig.yaml

```diff
@@ -31,9 +31,253 @@ spec:
+                          type: object
+                          additionalProperties: true
+                        requests:
+                          description: "Requests describes the minimum amount of compute resources required. If Requests is omitted for a container, it defaults to Limits if that is explicitly specified, otherwise to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/"
+                          type: object
+                          additionalProperties: true
+                    configResyncInterval:
+                      description: Resync interval for regular proxy broadcast updates
+                      type: string
+                    tlsMinProtocolVersion:
+                      description: The minimum TLS protocol version that the sidecar supports. Valid TLS protocol versions are TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2 and TLSv1_3.
+                      type: string
+                      enum:
+                        - TLS_AUTO
+                        - TLSv1_0
+                        - TLSv1_1
+                        - TLSv1_2
+                        - TLSv1_3
+                      default: TLSv1_2
+                    tlsMaxProtocolVersion:
+                      description: The maximum TLS protocol version that the sidecar supports. Valid TLS protocol versions are TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2 and TLSv1_3.
+                      type: string
+                      enum:
+                        - TLS_AUTO
+                        - TLSv1_0
+                        - TLSv1_1
+                        - TLSv1_2
+                        - TLSv1_3
+                      default: TLSv1_3
+                    cipherSuites:
+                      description: A list of ciphers that listener supports when negotiating TLS 1.0-1.2. This setting has no effect when negotiating TLS 1.3. For valid cipher names, see the latest OpenSSL ciphers manual page. E.g. https://www.openssl.org/docs/man1.1.1/apps/ciphers.html.
+                      type: array
+                      items:
+                        type: string
+                    ecdhCurves:
+                      description: A list of ECDH curves that TLS connection supports. If not specified, the curves are [X25519, P-256] for non-FIPS build and P-256 for builds using BoringSSL FIPS.
+                      type: array
+                      items:
+                        type: string
+                    proxyMode:
+                      description: Sets the destination ip address the envoy proxy will use when connecting to the backend application. Acceptable values are [LOCALHOST, POD_IP]. The default value is LOCALHOST
+                      type: string
+                      enum:
+                        - LOCALHOST
+                        - POD_IP
```

Comment:
lowercase for consistency with other values in custom resources?

Reply:
In that past, I've seen MACRO_CASE used to delimit enum fields in CRDs. If we have a different convention though, I can change it

Reply:
I saw that SMI CRDs use PascalCase, and I haven't seen any of our enums use MACRO_CASE.  https://github.com/openservicemesh/osm/blob/2f36980f85279ea8ddec80a261fe7bf76743648a/cmd/osm-bootstrap/crds/smi_traffic_access.yaml#L76-L78

In general, I would prefer to use lowercase or PascalCase for values.

Reply:
Scratch that [Kubernetes conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants) specify CamelCase for constant values so that's what I'll do